src/ie9-ajax.js

```diff
@@ -58,8 +58,17 @@ function ie9Ajax(method, url, opts) {
     xdr.onerror = handleError;
     xdr.ontimeout = handleError;
 
-    xdr.open(method, authToUrl(url, opts));
-    xdr.send(opts.data);
+    try {
+      xdr.open(method, authToUrl(url, opts));
+      xdr.send(opts.data);
+    } catch (error) {
+      // using trim because the error message has a bunch of white space for some reason
+      if (error.message.trim() === 'Access is denied.') {
+        reject(new Error('Please ensure Shopify JavaScript Buy SDK is run over https'));
```

Comment:
As looking at some examples of people using xdr and error handling. Some people might do something like this:
```
    function handleError(error) {
      reject(new Error(`There was an error with the XDR\n${JSON.stringify(error)}`));
    }
```